lib/ansible/modules/windows/win_get_url.ps1

```diff
@@ -19,120 +20,186 @@
 # WANT_JSON
 # POWERSHELL_COMMON
 
+
+Function CheckModified-File($url, $dest, $headers, $credentials, $timeout, $use_proxy, $proxy) {
```

Comment:
This is untested but I would have thought you would do this

```
if ($use_proxy) {
    if ($proxy_url -ne $null) {
        $proxy = New-Object System.Net.WebProxy($proxy_url, $true)
        if ($proxy_username -and $proxy_password) {
            $proxy_credential = New-Object System.Net.NetworkCredential($proxy_username, $proxy_password)
            $proxy.Credentials = $proxy_credential
        }
    } else {
        $proxy = [System.Net.WebRequest]::GetSystemWebProxy()
        $proxy.Credentials = [System.Net.CredentialCache]::DefaultCredentials
    }
} else {
    $proxy = $null
}
```
Then you just pass in the `$proxy` value and always assign that to the `Proxy` property of the WebClient.